pkg/datapath/linux/routing/routing.go

```diff
@@ -183,16 +187,36 @@ func Delete(ip net.IP, compat bool) error {
 	}
 
 	// Egress rules
-	egress := route.Rule{
-		Priority: priority,
-		From:     &ipWithMask,
-	}
-	if err := deleteRule(egress); err != nil {
-		return fmt.Errorf("unable to delete egress rule with ip %s: %v", ipWithMask.String(), err)
+	if info := node.GetRouterInfo(); info != nil && option.Config.IPAM == ipamOption.IPAMENI {
+		cidrs := make([]*net.IPNet, 0, len(info.GetIPv4CIDRs()))
+		for _, cidr := range info.GetIPv4CIDRs() {
+			c := cidr // reassign in order to be able to take a reference
+			cidrs = append(cidrs, &c)
+		}
+		// The CNI plugin coalesces the CIDR before installing the routes
+		routingCIDRs, _ := iputil.CoalesceCIDRs(cidrs)
```

Comment:
Should we factor this out into a common function shared between this code and the CNI plugin to ensure that they have the same behavior?

Reply:
As discussed offline, I instead added a comment mentioning that they need to kept in sync, as extracting common code is a bit tricky due to different Golang types.